transformer_engine/jax/cpp_extensions.py

```diff
@@ -351,31 +351,34 @@ def batcher(batched_args, batch_dims, *, zero_centered_gamma, epsilon):
     def infer_sharding_from_operands(zero_centered_gamma, epsilon, mesh, arg_infos, result_infos):
         del zero_centered_gamma, epsilon, result_infos
         x_spec = get_padded_spec(arg_infos[0])
-        assert x_spec[-1] is None, \
-            f"Does not support to shard hidden dim in {LayerNormFwdPrimitive.name}, " \
-            f"since it leads to incorrect results by missing necessary collective ops. " \
-            f"To address, you could apply jax.lax.with_sharding_constraint to force no " \
-            f"sharding on the hidden dim."
-        out_sharding = NamedSharding(mesh, PartitionSpec(*x_spec))
+        if x_spec[-1] is not None:
+            warnings.warn(
+                f"Does not support to shard hidden dim in {LayerNormFwdPrimitive.name}! " \
+                f"Force to not shard the hidden dim, which might introduce extra collective ops, " \
+                f"and hurt performance."
```

Comment:
What about this small variation?

```
                f"No native op support to shard hidden dim in {LayerNormFwdPrimitive.name}! " \
                f"So forcing to not shard the hidden dim, which might introduce extra collective ops, " \
                f"and hurt performance."
```
Otherwise, all is good.

Reply:
I assume the native XLA implementation supports the sharding on the hidden dims, but hurt perf a log.
It can be like:
```
calculate local mean of inputs -> allreduce_sum
calculate local mean of inputs's var -> allreduce_sum
then do remaining math.
```
Here would introduce two collectives ops, but it is supported. The reason custom_calls cannot support is we have no way to trigger collective ops in custom_calls, which will led to the wrong answer.